diff --git a/src/dedge.cpp b/src/dedge.cpp
index b030e20..fb85055 100644
--- a/src/dedge.cpp
+++ b/src/dedge.cpp
@@ -1,6 +1,8 @@
 #include "dedge.hpp"
 #include "config.hpp"
 
+#include <cstdint>
+
 #include <atomic>
 #include <fstream>
 #include <iostream>
diff --git a/src/dset.hpp b/src/dset.hpp
index f8e6f6d..66ce585 100644
--- a/src/dset.hpp
+++ b/src/dset.hpp
@@ -2,6 +2,7 @@
 #define __UNIONFIND_H
 
 #include <vector>
+#include <cstdint>
 #include <atomic>
 #include <iostream>
 
diff --git a/src/hierarchy.cpp b/src/hierarchy.cpp
index c333256..895ab90 100644
--- a/src/hierarchy.cpp
+++ b/src/hierarchy.cpp
@@ -1,4 +1,5 @@
 #include "hierarchy.hpp"
+#include <cstdint>
 #include <fstream>
 #include <algorithm>
 #include <unordered_map>
diff --git a/src/loader.cpp b/src/loader.cpp
index aa27066..00bb361 100644
--- a/src/loader.cpp
+++ b/src/loader.cpp
@@ -8,6 +8,8 @@
 
 #include "loader.hpp"
 
+#include <cstdint>
+
 #include <fstream>
 #include <unordered_map>
 
diff --git a/src/parametrizer-sing.cpp b/src/parametrizer-sing.cpp
index 2d600e3..2a004d3 100644
--- a/src/parametrizer-sing.cpp
+++ b/src/parametrizer-sing.cpp
@@ -1,4 +1,5 @@
 #include "config.hpp"
+#include <cstdint>
 #include "field-math.hpp"
 #include "parametrizer.hpp"
 
